File: TwoPointers/LinkedList/Leetcode83.java
public class Leetcode83 {
     // Leetcode 83
    /**
     * This one is basically the same, just we are remove linkedlist
     * @param head
     * @return
     */
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null)return null;
        ListNode dummy = new ListNode();
        dummy.next = head;
        ListNode slow = head, fast = head;
        while(fast!= null){
            if(fast.val == slow.val){
                fast = fast.next;
            }else{
                slow.next =fast;
                slow = slow.next;
            }
        }
        // remember to do this steps since we do not want the following node.
        slow.next = null;
        return dummy.next;
    }
}

File: LinkedList/Python/Leetcode83.py
class Leetcode83:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        slow = fast = head
        def findNext(left, right):
            while right and left.val == right.val:
                right = right.next
            return right
        while slow and slow.next:
            nextNode = findNext(slow, fast)
            slow.next = nextNode
            slow = fast = nextNode
        return head
